playlist_souffle/use_cases/souffle_playlist.py

```diff
@@ -0,0 +1,39 @@
+"""Module for the souffle_playlist use case."""
+
+from datetime import datetime as dt
+from playlist_souffle.definitions.playlist import Playlist
+import playlist_souffle.entities.souffler as souffler_entity
+import playlist_souffle.entities.track as track_entity
+
+def souffle_playlist(spotify, playlist_uri, user_id, shuffle_by):
+    """Souffle a playlist."""
+
+    original_playlist = Playlist(
+        owner=None,
+        name=spotify.fetch_playlist_name(playlist_uri),
+        tracks=spotify.fetch_playlist_tracks(playlist_uri),
+        description=None
+    )
+
+    collection_id_by_track = {track: track_entity.extract_track_collection_id(track, shuffle_by)
+                              for track in original_playlist.tracks}
+
+    collection_tracks_by_collection_id = spotify.fetch_collection_tracks_by_collection_id(
```

Comment:
@evanhammer: do you think this and the following line creating ```collection_tracks_by_track``` should be combined into one gateway call?

```py
def fetch_collection_tracks_by_track(collection_id_by_track):
    """Fetch collection tracks by track given a collection_id_by_track map."""
```

Just would be moving some business logic from use case to gateway.